ext/semian/resource.c

```diff
@@ -152,6 +155,62 @@ cleanup_semian_resource_acquire(VALUE self)
   return Qnil;
 }
 
+static long check_permissions_arg(VALUE permissions)
```

Comment:
It's custom in Semian (and Rubby C-extensions, since Ruby follows this convention) to have the type on a line before the function name